Adding database

R0DR160HM opened the pull request:
Adding `database` package
https://hexdocs.pm/database/

lpil commented:
Hello! Thank you for this, but I'm a bit reluctant to add it to the list as it doesn't have any documentation and it looks like it's not type safe. You can parameterise a table with a type that isn't an Erlang record and I think it will crash, and you can open an existing database table with the wrong type and it won't detect that.

R0DR160HM commented:
Hello, thank you for the feedback!

I've improved the type checks, and addressing your concerns:
1. Added documentation to the module and all public functions/types.
2. It now validates whether the provided table definition is a record
3. Now it detects if you try to open a table with the wrong type, even if they have the same name.
